code/code/sys/configuration.cc

```diff
@@ -183,3 +186,30 @@ bool Config::doConfiguration(int argc, char* argv[]) {
   }
   return true;
 }
+
+std::string readFileContents(const std::string& path) {
```

Comment:
Is there a reason why this is not in an anonymous namespace?

Reply:
Also, I'm confused - is this meant to return a string or an istringstream?

Reply:
What is the reason you split the reading into a different function than the parsing?

Reply:
> Is there a reason why this is not in an anonymous namespace?
> What is the reason you split the reading into a different function than the parsing?

"Don't let good code go to waste", basically. 😄 I figured it was a useful function that could be used elsewhere in the code in the future, so might as well make it accessible. Could probably find a better place for it than `configuration.h/cc` I suppose, though not sure where that would be off the top of my head.

We do have a `file_to_sstring` function defined elsewhere that does something similar, but using much older/messier syntax and no RAII types. I tested short-circuiting that function by just calling my `readFileContents` function instead and it worked mostly fine, though needs more testing and a few tweaks, so I figured I'd do that in a separate PR sometime.

> Also, I'm confused - is this meant to return a string or an istringstream?

Not really sure why I did it this way now that I look at it again. Probably a leftover from how I had the `readVersionFromFile` function coded previously (it went through several iterations). But you're right, I do just convert it to an `istringstream` immediately after returning the value as a `std::string`, which seems silly. I'll change that.

Reply:
> "Don't let good code go to waste"

There's also the YAGNI principle... Oh, yeah, I wanted but forgot to ask about `file_to_sstring`. If it's actually going to be used, feel free to retain it.

Reply:
First I've heard of YAGNI, somehow, but I like it.

Reply:
Fixed in https://github.com/sneezymud/sneezymud/pull/476/commits/17de1b40cc65e5f489f434878ab0267af991d881